Improved share layout: centered app, darkened background, and extended footer full-width

aoby opened the pull request:
New look:
![image](https://cloud.githubusercontent.com/assets/8405316/11642574/6b9b6724-9cf3-11e5-9017-ff34932163fb.png)

![image](https://cloud.githubusercontent.com/assets/8405316/11642584/89bc9ee4-9cf3-11e5-9b73-851fa57dfc0d.png)

Bjvanminnen commented on dashboard/app/views/layouts/_share_footer.html.haml:
```diff
@@ -8,7 +8,7 @@
     menuItems: [
       {
-        text: I18n.t('footer.make_my_own_app'),
+        text: I18n.t('footer.try_hour_of_code'),
```
Do we intend to change this string on all shares, i.e. including app lab?

aoby replied:
No. Applab doesn't use this _share_footer layout, because it generates its footer on the client side: https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/applab/applab.js#L346-L346

Bjvanminnen commented:
This makes speed slider icons hard to see on artist. Might want to make those lighter.
![image](https://cloud.githubusercontent.com/assets/1767466/11574414/6c7a5136-99c0-11e5-9f24-87cb42feb7fc.png)

Bjvanminnen commented:
Might be something we're fine with, but for playlab's default background, the fact that it's also dark (i.e. playspace background and page background end up both being dark, but different) makes it not look as good on the share page imo. Not sure if/how much we care. Perhaps a question for PM

![image](https://cloud.githubusercontent.com/assets/1767466/11574518/f7743946-99c0-11e5-9e18-d6a273ff1d42.png)

aoby commented:
@kozzi @alicesteinglass, see Brent's playlab screenshot above.

kozzi commented:
I'm fine with it. Once we get the frame in (likely post HOC), I think it'll feel better.

aoby commented:
@Bjvanminnen The white icons look a bit pixelized, but they do show up better against the background:

![screen shot 2015-12-03 at 5 01 01 pm](https://cloud.githubusercontent.com/assets/8405316/11579060/93d4f7f8-99df-11e5-80bc-0557d36b878f.png)

Bjvanminnen commented:
Hmmm....that doesn't look like the right image then. Will double check
On Dec 3, 2015 5:03 PM, "Andrew Oberhardt" notifications@github.com wrote:

> @Bjvanminnen https://github.com/Bjvanminnen The white icons look a bit
> pixelized, but they do show up better against the background:
> 
> [image: screen shot 2015-12-03 at 5 01 01 pm]
> https://cloud.githubusercontent.com/assets/8405316/11579060/93d4f7f8-99df-11e5-80bc-0557d36b878f.png
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/code-dot-org/code-dot-org/pull/5850#issuecomment-161841634
> .

aoby commented:
I updated share view to use the white icons (though we still need to update the png) and added a Try an Hour of Code header that links to code.org/learn:

![screen shot 2015-12-03 at 5 09 29 pm](https://cloud.githubusercontent.com/assets/8405316/11579228/fdeb5b40-99e0-11e5-9637-d58520d815f4.png)

aoby commented:
@philbogle we want to get these share layout changes in for HOC. Let me know when / if this is ok to merge.

aoby commented:
New image looks good: 
![screen shot 2015-12-03 at 5 21 13 pm](https://cloud.githubusercontent.com/assets/8405316/11579646/882d8794-99e4-11e5-8515-1dd7ec156b9f.png)

alicesteinglass commented:
As for the default playlab background creating a dark on dark look, I wouldn't worry about - almost everything looks better on dark and the default backgrounds for the two new playlab themes are not dark (and my guess is they'll be more popular than classic). We can change the default background for playlab as a follow up.

bcjordan commented on apps/src/craft/craft.js:
```diff
@@ -281,6 +281,10 @@ Craft.init = function (config) {
+
+      // Adjust visualizationColumn width.
+      var visualizationColumn = document.getElementById('visualizationColumn');
+      visualizationColumn.style.width = this.nativeVizWidth + 'px';
```
Would this override the CSS browser-window-dimension-based sizing from happening pre-handle-grab? e.g. open page, drag window size up and down, should change size based on browser window size.

If it does, could potentially make this only happen when on share page

aoby replied:
Good catch. You're right, this messed with resizing in the non-sharing view. I updated this to only apply to sharing.

philbogle replied:
Looking at this now.

On Fri, Dec 4, 2015 at 12:47 AM, Andrew Oberhardt notifications@github.com
wrote:

> In apps/src/craft/craft.js
> https://github.com/code-dot-org/code-dot-org/pull/5850#discussion_r46660033
> :
> 
> > @@ -281,6 +281,10 @@ Craft.init = function (config) {
> >        if (!config.level.showPopupOnLoad) {
> >          Craft.initializeAppLevel(config.level);
> >        }
> > +
> > -      // Adjust visualizationColumn width.
> > -      var visualizationColumn = document.getElementById('visualizationColumn');
> > -      visualizationColumn.style.width = this.nativeVizWidth + 'px';
> 
> Good catch. You're right, this messed with resizing in the non-sharing
> view. I updated this to only apply to sharing.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/code-dot-org/code-dot-org/pull/5850/files#r46660033.

philbogle commented:
The test matrix is non-trivial (with all the nested conditionals) and I don't have the background or spec definition to say whether it's all correct and doesn't regress anything; I think any regressions will have to be caught via manual testing.

Could you write out the different test cases we should cover?  If this change is deemed truly essential, and we can ensure there are no regressions in manual testing, then I'm OK with this going in.

aoby commented:
Test matrix:
- share view
  - legacy - verify tryHoc banner, dark background, centered app, wide share footer
    - minecraft
    - starwars
    - artist
    - playlab
    - flappy
  - applab (unchanged)
- normal, non-share view for all of the above (unchanged)

@Bjvanminnen please double check this. I may be missing a relevant app type

Project backed artist
Project backed playlab
Frozen artist share (i.e. level 6 of frozen script)

Bjvanminnen commented:
I would add project-based artist and playlab to the list (i.e. share from /s/artist/stage/1/puzzle/10 for the legacy view and from /projects/artist for the project-backed view)

breville commented:
Is "how it works" hidden on desktop inside the little footer?  If so, I am super disappointed in that change.  The magic of /gallery is going through shared levels, playing them, and then pressing "how it works" to see how they work.  On staging right now, in a desktop browser, nobody's going to find that option buried in the little popup footer at the very bottom.  I strongly urge reconsidering this.  fyi @alicesteinglass and @kozzi

kozzi commented:
I don't believe we changed the design with this work. I think "how it works" was always inside the pop up menu. 

That said, not saying what was already there was good. We can look at it post HOC. We just want people to try their own Hour of Code more than poke around in someone else's (at least for next week). It's definitely more interesting with App Lab shares. 

Might also be good to look at data around how often people actually click "how it works".

breville commented:
No, clicking into a level from gallery on desktop showed a big "how it works" button until a few days ago.  In fact, Roxanne asked me to write a blog post about this for this/next week, but with the button gone I'm worried it won't be worth doing.  In my mind this is a regression.

Bjvanminnen commented:
Agree with Brendan that this is def something that changed (we used to have a big how it works button).  Assumed this was by design. I think we would have to feel very strongly about this to change it back before HOC.

breville commented:
Brent's right.  That said, I have a blog post ready to go which encourages students/teachers to use that exact button to use the star wars tutorial to its full potential.  If it were as simple as restoring a single button (which looks like a one-liner fix after https://github.com/code-dot-org/code-dot-org/pull/5742) then I'd request it to be considered.

kozzi commented:
Oh I misremembered. I was thinking of the app lab shared pages (which did already have "How it works" inside a little footer). Andrew's work was to update the HOC tutorial shared pages to copy the app lab shared pages.

I assumed that since app lab shares had moved "how it works" into the footer (and app lab is actually the more interesting place to have the "how it works" button), we'd be okay with doing this for HOC tutorials. 

That said, I don't have a lot of history here. I'm still inclined to not change it back for HOC, since reverting the change will put all the cruft back into the shared pages which looks very broken and unpolished :( 

@alicesteinglass thoughts?

Bjvanminnen commented:
The code to add the button back might be quite simple, but the test matrix (as mentioned a couple posts earlier) is not. We're getting very last minute at this point.

Andrew is going to look and see how easy it would be to add this button back if we wanted, but my inclination is that it's too late to change things at this point.

caleybrock commented:
I'm a bit concerned about just leaving the current state as is. I actually thought it was broken without a header/lack of spacing. I also didn't realize the bottom bar was actionable so I couldn't find 'How it works'. The 'How it works' call to action is a big plus of the gallery flow for driving users to get inspired from each other's code.
<img width="1196" alt="screen shot 2015-12-03 at 11 59 15 pm" src="https://cloud.githubusercontent.com/assets/4640747/11596938/06889d76-9a6d-11e5-98dc-7c0b450488b8.png">
